app/lib/cashu/wallet.ts

```diff
@@ -0,0 +1,209 @@
+import type {
+  CashuWallet,
+  MeltQuoteResponse,
+  MintQuoteResponse,
+  Proof,
+} from '@cashu/cashu-ts';
+import type {
+  MeltProofOptions,
+  MintProofOptions,
+  SwapOptions,
+} from '@cashu/cashu-ts';
+import { sumProofs } from './proof';
+
+type GetCrossMintQuotesResult = {
+  /** Mint quote for the destination wallet */
+  mintQuote: MintQuoteResponse;
+  /** Melt quote for the source wallet */
+  meltQuote: MeltQuoteResponse;
+};
+
+type SwapProofsResult = {
+  /** New proofs from the destination wallet that can be claimed */
+  newProofs: Proof[];
+  /** Change leftover from the swap */
+  change: Proof[];
+};
+
+/**
+ * Attempts to find valid quotes for cross-mint swaps by iteratively adjusting the amount
+ * to account for fees. A valid quote is found when the total cost (amount + fees) is
+ * less than or equal to the available proofs.
+ *
+ * @param from - Source wallet to melt proofs from
+ * @param to - Destination wallet to mint new proofs to
+ * @param proofsAvailable - Total amount available in source proofs
+ * @param requestedAmount - The amount of proofs to mint
+ * @param errorTolerance - Maximum difference between estimated and actual fee reserve (default: 2)
+ * @param maxAttempts - Maximum number of attempts to find valid quotes (default: 5)
+ */
+export const getCrossMintQuotes = async (
+  from: CashuWallet,
+  to: CashuWallet,
+  proofsAvailable: number,
+  requestedAmount: number,
+  errorTolerance = 2,
+  maxAttempts = 5,
+): Promise<GetCrossMintQuotesResult> => {
+  if (requestedAmount > proofsAvailable) {
+    throw new Error(
+      `Requested amount is greater than available proofs: ${requestedAmount} > ${proofsAvailable}`,
+    );
+  }
+
+  // TODO: convert from.unit to to.unit and remove this check
+  // will require an exchange rate
+  if (from.unit !== to.unit) {
+    throw new Error('Cannot swap between different units');
+  }
+
+  // Estimate initial fee based on Nutshell defaults to limit the number of attempts:
+  // See: https://github.com/cashubtc/nutshell/blob/main/cashu/core/helpers.py#L53-L58
+  const estimatedFeeReserve = Math.max(2, Math.floor(requestedAmount * 0.1));
```

Comment:
I am not sure I understand this. In the linked code they are calculating fee reserve using those values form settings. where can I see those settings?

Reply:
0.01 here